server/schedulers/hot_region.go

```diff
@@ -201,8 +201,7 @@ const balanceHotRetryLimit = 10
 
 func (h *balanceHotRegionsScheduler) balanceHotWriteRegions(cluster opt.Cluster) []*operator.Operator {
 	for i := 0; i < balanceHotRetryLimit; i++ {
-		switch h.r.Int() % 2 {
-		case 0:
+		if h.allowBalanceRegion(cluster) && (!h.allowBalanceLeader(cluster) || h.r.Int()%2 == 0) {
```

Comment:
Is it necessary `!h.allowBalanceLeader(cluster) `?

Reply:
Yes. Without it, when `allowbalanceLeader = false` and `r.Int()%2=0`, it won't create any operator.